src/Microsoft.VisualStudio.ProjectSystem.Managed/Tree/ProjectImportsSubTreeProvider.cs

```diff
@@ -0,0 +1,280 @@
+
+                                        tree = tree.Add(newChild).Parent!;
+                                    }
+                                }
+
+                                return tree;
+                            }
+                        }
+
+                        static IDisposable TrySuppressExecutionContextFlow()
+                        {
+                            return ExecutionContext.IsFlowSuppressed()
+                                ? EmptyDisposable.Instance
+                                : ExecutionContext.SuppressFlow();
+                        }
+                    }
+
+                    void TearDownTree()
+                    {
+                        Assumes.NotNull(_subscription);
+                        _subscription.Dispose();
+                        _subscription = null;
+
+                        SubmitTreeUpdateAsync(
+                            (currentTree, a, c) =>
+                            {
+                                IProjectTree tree = currentTree.Value.Tree;
+
+                                // Set invisible
+                                tree = tree.SetVisible(false);
+
+                                // Remove all children
+                                while (tree.Children.Count != 0)
+                                {
+                                    tree = tree.Children[0].Remove();
+                                }
+
+                                return Task.FromResult(new TreeUpdateResult(tree));
+                            });
+                    }
+                }
+            }
+        }
+
+        // CPS's physical tree provider currently only uses the IShowAllFilesProjectTreeProvider.ShowAllFiles property
+        // for graft root providers. These other interface methods have safe no-op implementations in case that changes
+        // one day.
+
+        bool IShowAllFilesProjectTreeProvider.CanIncludeItems(IImmutableSet<IProjectTree> nodes)   => true;
+        bool IShowAllFilesProjectTreeProvider.CanExcludeItems(IImmutableSet<IProjectTree> nodes)   => true;
+        Task IShowAllFilesProjectTreeProvider.IncludeItemsAsync(IImmutableSet<IProjectTree> nodes) => Task.CompletedTask;
+        Task IShowAllFilesProjectTreeProvider.ExcludeItemsAsync(IImmutableSet<IProjectTree> nodes) => Task.CompletedTask;
+
+        protected override ConfiguredProjectExports GetActiveConfiguredProjectExports(ConfiguredProject newActiveConfiguredProject)
+        {
+            Requires.NotNull(newActiveConfiguredProject, nameof(newActiveConfiguredProject));
+
+            return GetActiveConfiguredProjectExports<MyConfiguredProjectExports>(newActiveConfiguredProject);
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _subscription?.Dispose();
+            }
+
+            base.Dispose(disposing);
+        }
+
+        [Export]
+        [AppliesTo(Capability)]
```

Comment:
There's a race here that can cause this AppliesTo to be out of sync with the AppliesTo on the provider. Underneath this capabiliity will be respected by the GetExportedValue call, which will respect whatever the current set of capabilities for the project are, which may or may not have changed from the set of capabilities that caused this tree to be created, causing it to return null for this component. I would just remove the capability - if the tree exists, so should this export.